### PR TITLE
Pairwise distances

### PR DESCRIPTION
**Pull request by msluszniak**

### CODE REVIEW

File: lib/scholar/manifold/tsne.ex [hunk 1]
```diff
@@ -167,18 +167,21 @@ defmodule Scholar.Manifold.TSNE do
...
       :euclidean ->
         Distance.euclidean(t1, t2, axes: [2])
+        # Distance.pairwise_euclidean(x)
 
       :manhattan ->
         Distance.manhattan(t1, t2, axes: [2])
 
       :cosine ->
         Distance.cosine(t1, t2, axes: [2])
+        # Distance.pairwise_cosine(x)
```
Comment (msluszniak) [inline lib/scholar/manifold/tsne.ex:184]: I have no idea why using pairwise distances causes the results to be completely different than those from broadcasting and calculating distances by hand. I tested this function, and it returns the same results for the same input value.

**Review by josevalim — APPROVED** [review]

File: lib/scholar/manifold/tsne.ex (hunk 1 above)
Comment (josevalim) [reply lib/scholar/manifold/tsne.ex:184]: Try doing `print_expr` and see if the expressions are the same or different?

Comment (msluszniak) [reply lib/scholar/manifold/tsne.ex:184]: Okay, I discovered that after 10 iterations of tsne algorithms, both functions gives the same results. After 50, there is a small discrepancy in distances, and after 500 iterations (default) there are completely different results.

**Comment by msluszniak** [comment]
I tested TSNE with pairwise distances and despite results from pairwise and classical distances generated results with different orders of magnitude, it seems to work. I plotted the results for Iris, and they are reasonable. For now, I changed the test so they are passing, but we may think of more sophisticated tests that will pass if embedded space is "good enough".

**Review by josevalim — APPROVED** [review]

**Pull request merged**